pkg/providers/vsphere/reconciler/reconciler_test.go

```diff
@@ -149,6 +150,26 @@ func TestReconcileCNIErrorClientRegistry(t *testing.T) {
 	tt.Expect(result).To(Equal(controller.Result{}))
 }
 
+func TestReconcilerReconcileControlPlaneSuccess(t *testing.T) {
+	tt := newReconcilerTest(t)
+	tt.createAllObjs()
+
+	result, err := tt.reconciler().ReconcileControlPlane(tt.ctx, test.NewNullLogger(), tt.buildSpec())
+
+	tt.Expect(err).NotTo(HaveOccurred())
+	tt.Expect(tt.cluster.Status.FailureMessage).To(BeZero())
+	tt.Expect(result).To(Equal(controller.Result{}))
+
+	obj := &addonsv1.ClusterResourceSet{ObjectMeta: metav1.ObjectMeta{
+		Name:      "workload-cluster-crs-0",
+		Namespace: "eksa-system",
+	}}
+	key := client.ObjectKeyFromObject(obj)
+	tt.Eventually(func() error {
+		return tt.client.Get(tt.ctx, key, obj)
+	}).Should(Succeed(), "\"object %s should eventually exist", obj.GetName())
```

Comment:
Can we do the same thing for the other CP objects? at least the obvious ones: capi cluster, kcp and vsphere machine template
I'm approving the PR, I think you can add it as a follow up PR :)